dolphinscheduler-worker/src/main/java/org/apache/dolphinscheduler/server/worker/WorkerServer.java

```diff
@@ -197,6 +202,8 @@ public void close(String cause) {
                 logger.warn("thread sleep exception", e);
             }
 
+            this.killAllRunningTasks();
```

Comment:
Maybe put `this.killAllRunningTasks()` after other close might be better? What do you think?
```java
this.nettyRemotingServer.close();
this.workerRegistryClient.unRegistry();
this.alertClientService.close();
this.springApplicationContext.close();
this.killAllRunningTasks();
```

Reply:
It seems to be possible, but I think the close of the spring component is best placed at the end.

Reply:
It Is OK to put the `this.springApplicationContext.close()` at the end.
I just worry that the worker will receive a new request before `this.killAllRunningTasks()`, so put `this.killAllRunningTasks()` after `this.nettyRemotingServer.close()` is fine.

Reply:
After `Stopper.stop()` , the stop signal is true. Although worker can receive a new request, but it will not submit `taskExecuteThread`.
```
while (Stopper.isRunning()) {
        try {
            taskExecuteThread = workerExecuteQueue.take();
            workerExecService.submit(taskExecuteThread);
        } catch (Exception e) {
            logger.error("An unexpected interrupt is happened, "
                + "the exception will be ignored and this thread will continue to run", e);
        }
}
```

Reply:
OK, I got you. But it will be great if we can simply change the close order. In the future, if someone modified the request submit logic or stop logic, it should also work well.

Reply:
I had adjusted the orders of close. @ruanwenjun Please take a look again and thanks for your good suggestion.